src/Serilog/Settings/KeyValuePairs/KeyValuePairSettings.cs

```diff
@@ -106,6 +118,19 @@ public void Configure(LoggerConfiguration loggerConfiguration)
                 }
             }
 
+            var declaredVariables = ParseVariableDeclarationDirectives(directives);
+
+            string minimumLevelControlledByLevelSwitchName;
+            if (directives.TryGetValue(MinimumLevelControlledByDirective, out minimumLevelControlledByLevelSwitchName))
+            {
+                var levelSwitch = LookUpVariable(minimumLevelControlledByLevelSwitchName, typeof(LoggingLevelSwitch), declaredVariables) as LoggingLevelSwitch;
+                // TODO : maybe throw an exception to notify missing variable declaration ?
```

Comment:
I'm not sure what is the *policy* around exception in Serilog. 
I know the idea is that a logging library shouldn't crash the app, but is it ok to throw exceptions during configuration, to help the developer fix the incorrect config ? 
I believe it would be more user friendly to throw an exception that states what is exactly wrong (undeclared variable / wrong name etc), but I was not sure :p

Reply:
https://github.com/serilog/serilog/wiki/Reliability should have some clues around this (it's a little vague, but sets out the general idea).

Reply:
thanks ! I knew I had read something along those lines somewhere, I just couldn't remember :)
I'll change the code to throw a more explicit exception